caffe2/python/pybind_state.cc

```diff
@@ -1673,6 +1673,7 @@ void addGlobalMethods(py::module& m) {
       "onnxifi",
       [](const py::bytes& pred_net_str,
          const std::unordered_map<std::string, std::vector<int>>& shapes,
+         bool infer_shapes,
```

Comment:
what's the reason of not always doing shape inference?

Reply:
Sometimes shape inference is not feasible as the output shape depends not only the input shape but also the input value. One example is those SparseNN ops. In this case, we have to do a fake run. 

Another reason is that our shape inference is not very robust. It might fail even it's not the case above.

Reply:
I see. I think the first reason shouldn't prevent us to always set infer_shapes to true? (while the second one is the real blocker).

Reply:
Actually I think the second one is the real blocker. If the shape inference is buggy we can still fix. But if there is no way to infer shape just from the input shape, we have to do a RunNet() to populate the shapes.